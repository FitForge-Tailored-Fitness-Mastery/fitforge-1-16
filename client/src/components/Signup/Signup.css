.signup-container {
    width: auto;
    height: 100vh;
    position: relative;
    background-image: url('./loginbg.jpg'); 
    background-size: cover; /* Cover the entire container */
    background-position: center; /* Center the background image */
    display: flex;
    justify-content:space-around;
    align-items: center;
    margin: 0px;
    padding: 0px;
    color: white;
}
  
.signup-container::before {
    content: ''; /* Required for pseudo-elements */
    position: absolute; /* Position relative to the login-container */
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: rgba(0, 0, 0, 0.7); /* Black background with 50% opacity */
    z-index: 0; /* Ensure it's below the content but above the background image */
}

.signup-box{
    max-width: 400px;
    width: 100%;
    padding: 20px;
    margin: 10px;
    background-color: rgba(0, 0, 0, 0.4);    
    border-radius: 10px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
    position: relative;
    z-index: 1; /* Higher than the overlay */
}

.signup-form {
    display: flex;
    flex-direction: column;
}

.signup-form > h1 {
    color: #fdd673;
    text-align: center;
    font-size: 2.3rem;
    margin-bottom: 0px;
}

.signup-form .signup-tagline{
  font-size: 0.8rem;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  margin-bottom: 35px;
  font-weight: 100;
}

.signup-form label {
    display: block;
    margin-bottom: 5px;
    color: white;
}
  
.signup-form input {
    width: 90%;
    margin-bottom: 8px;
    padding: 7.5px;
    border: none;
    border-radius: 5px;
    font-size: 0.9rem;
}
  
.signup-button {
    margin-top: 25px;
    padding: 7px;
    display: inline-block;
    align-self: center;
    background-color: #fdd673 ;
    color: black;
    border: none;
    border-radius: 30px;
    cursor: pointer;
    font-weight: 650;
    font-size: 1.2rem;
    width: 50%;
  }

.login-link{
    margin-top: 15px;
    text-align: center;
    color: white;
}

.login-link a {
    color: #fdd673;
    text-decoration: none;
}

.login-link a:hover {
  text-decoration: underline;
}
  