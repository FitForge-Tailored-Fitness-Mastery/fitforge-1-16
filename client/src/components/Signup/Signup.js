import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

import axios from 'axios';
import './Signup.css'; // Make sure to create this CSS file

const Signup = () => {
  const [formData, setFormData] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
  });
  
  const navigate = useNavigate();

  const handleChange = (e) => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (formData.password !== formData.confirmPassword) {
      alert("Passwords don't match.");
      return;
    }
  
    try {
      // Make sure the URL matches the endpoint from your Express server
      const response = await axios.post('http://localhost:5000/signup', formData);
      console.log(response.data);
      // Handle success (redirect or show a message)
      navigate('/client-home');
    } catch (error) {
      console.error('Signup error:', error.response.data);
      // Handle errors (show error message to user)
    }
  };
  

  return (
    <div className="signup-container">
      <form className="signup-form" onSubmit={handleSubmit}>
        <h1>FitForge</h1>
        <h2 className="signup-title">Sign Up</h2>
        <p className="signup-tagline">Fitness is a click away</p>
        <label htmlFor="firstName">First Name</label>
        <input
          id="firstName"
          type="text"
          name="firstName"
          value={formData.firstName}
          onChange={handleChange}
          required
        />
        <label htmlFor="lastName">Last Name</label>
        <input
          id="lastName"
          type="text"
          name="lastName"
          value={formData.lastName}
          onChange={handleChange}
          required
        />
        <label htmlFor="email">Email</label>
        <input
          id="email"
          type="email"
          name="email"
          value={formData.email}
          onChange={handleChange}
          required
        />
        <label htmlFor="password">Password</label>
        <input
          id="password"
          type="password"
          name="password"
          value={formData.password}
          onChange={handleChange}
          required
        />
        <label htmlFor="confirmPassword">Confirm Password</label>
        <input
          id="confirmPassword"
          type="password"
          name="confirmPassword"
          value={formData.confirmPassword}
          onChange={handleChange}
          required
        />
        <button type="submit" className="signup-button">Sign Up</button>
      </form>
    </div>
  );
};

export default Signup;
