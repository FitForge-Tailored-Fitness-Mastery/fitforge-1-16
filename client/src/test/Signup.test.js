import React from 'react';
import { render, fireEvent, screen, act } from '@testing-library/react';
import axios from 'axios'; // Mock axios for testing

import Signup from '../components/Signup/Signup';

jest.mock('axios');

describe('Signup component', () => {
  test('renders sign-up form', () => {
    render(<Signup />);

    expect(screen.getByText('FitForge')).toBeInTheDocument();
    expect(screen.getByLabelText('First Name')).toBeInTheDocument();
    expect(screen.getByLabelText('Last Name')).toBeInTheDocument();
    expect(screen.getByLabelText('Email')).toBeInTheDocument();
    expect(screen.getByLabelText('Password')).toBeInTheDocument();
    expect(screen.getByLabelText('Confirm Password')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Sign Up' })).toBeInTheDocument();
  });

  test('updates form data on input change', () => {
    render(<Signup />);

    const firstNameInput = screen.getByLabelText('First Name');
    const lastNameInput = screen.getByLabelText('Last Name');
    const emailInput = screen.getByLabelText('Email');
    const passwordInput = screen.getByLabelText('Password');
    const confirmPasswordInput = screen.getByLabelText('Confirm Password');

    fireEvent.change(firstNameInput, { target: { value: 'John' } });
    fireEvent.change(lastNameInput, { target: { value: 'Doe' } });
    fireEvent.change(emailInput, { target: { value: 'john@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'password123' } });

    expect(firstNameInput.value).toBe('John');
    expect(lastNameInput.value).toBe('Doe');
    expect(emailInput.value).toBe('john@example.com');
    expect(passwordInput.value).toBe('password123');
    expect(confirmPasswordInput.value).toBe('password123');
  });

  test('submits sign-up form', async () => {
    axios.post.mockResolvedValueOnce({ data: { message: 'User created successfully' } });

    render(<Signup />);

    const signUpButton = screen.getByRole('button', { name: 'Sign Up' });

    await act(async () => {
      fireEvent.click(signUpButton);
      await new Promise((resolve) => setTimeout(resolve, 0)); // Wait for promises to resolve
    });

    expect(axios.post).toHaveBeenCalledTimes(1);
    expect(axios.post).toHaveBeenCalledWith('http://localhost:5000/signup', {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      confirmPassword: '',
    });
  });
});
