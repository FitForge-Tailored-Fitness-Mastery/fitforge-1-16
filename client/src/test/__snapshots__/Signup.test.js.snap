// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component testing the sign up feature 1`] = `
<div
  className="signup-container"
>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <h1>
      FitForge
    </h1>
    <h2
      className="signup-title"
    >
      Sign Up
    </h2>
    <p
      className="signup-tagline"
    >
      Fitness is a click away
    </p>
    <label
      htmlFor="firstName"
    >
      First Name
    </label>
    <input
      id="firstName"
      name="firstName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="lastName"
    >
      Last Name
    </label>
    <input
      id="lastName"
      name="lastName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <label
      htmlFor="confirmPassword"
    >
      Confirm Password
    </label>
    <input
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <button
      className="signup-button"
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
